pytket/qir/conversion/api.py

```diff
@@ -122,12 +133,42 @@ def pytket_to_qir(
             qir_int_type=int_type,
             trunc=trunc,
         )
+    elif profile == QIRProfile.AZUREADAPTIVE:
+        qir_generator = AzureAdaptiveProfileQirGenerator(
+            circuit=circ,
+            module=m,
+            wasm_int_type=int_type,
+            qir_int_type=int_type,
+            trunc=trunc,
+        )
     else:
         raise NotImplementedError("unexpected profile")
 
     populated_module = qir_generator.circuit_to_module(qir_generator.circuit, True)
 
-    if qir_generator.has_wasm:
+    if profile == QIRProfile.AZUREADAPTIVE:
+
+        assert not qir_generator.has_wasm
+
+        sar_azure_dict: dict[str, str] = qir_generator.get_azure_sar()
+
+        initial_result = str(populated_module.module.ir())
+
+        for az in sar_azure_dict:
+            initial_result = initial_result.replace(az, sar_azure_dict[az])
+
+        result = initial_result
+
+        bitcode = pyqir.Module.from_ir(pyqir.Context(), result).bitcode
```

Comment:
May as well only compute this if `qir_format == QIRFormat.BINARY`.

Reply:
I think it would be good to keep that. That we we make sure that our search and replace is still parable for pyqir and we don't submit anything invalid.